### PR TITLE
Limit ShellBolt internal _pendingWrites queue length

### PR DESCRIPTION
**Pull request by xiaokang**
for storm issuse #541

### CODE REVIEW

File: storm-core/src/jvm/backtype/storm/task/ShellBolt.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
     private ShellProcess _process;
     private volatile boolean _running = true;
     private volatile Throwable _exception;
-    private LinkedBlockingQueue _pendingWrites = new LinkedBlockingQueue();
+    private LinkedBlockingQueue _pendingWrites = new LinkedBlockingQueue(100);
```
Comment (nathanmarz) [inline storm-core/src/jvm/backtype/storm/task/ShellBolt.java]: This shouldn't be initialized here if its going to be initialized in the constructor.

File: storm-core/src/jvm/backtype/storm/task/ShellBolt.java [hunk 2]
```diff
@@ -66,7 +66,12 @@ public ShellBolt(ShellComponent component) {
     }
 
     public ShellBolt(String... command) {
+        this(100, command);
+    }
+    
+    public ShellBolt(int maxQueueSize, String... command) {
         _command = command;
+        _pendingWrites = new LinkedBlockingQueue(maxQueueSize);
```
Comment (nathanmarz) [inline storm-core/src/jvm/backtype/storm/task/ShellBolt.java]: This should be moved to prepare and be made configurable via the Storm config, not via constructor params.

**Comment by nathanmarz** [comment]
Made some comments about changes that should be made. These are important because from a pure multilang definition, ShellBolt isn't constructed by the user so they need to be able to configure it via config.

**Comment by xiaokang** [comment]
hi @nathanmarz , thanks for your comments. I adjust the code to config max pending in ShellBolt's prepare.

**Comment by nathanmarz** [comment]
Looks like this needs to be rebased against master, as it can't be merged right now.

**Comment by xiaokang** [comment]
Just rebased the branch against master but other's commits appear in this PR. Is it ok?

**Comment by nathanmarz** [comment]
Hmm, that's really strange. That's definitely not ok. Maybe you can just close this one and open a new one against master?

**Comment by xiaokang** [comment]
OK, #730 sent. Close this one.

**Pull request closed without merging**